recipe/meta.yaml

```diff
@@ -1,75 +1,137 @@
-    - typed-ast >=1.4.0,<2  # [py<38]
-    - typing_extensions >=3.10
-    - tomli >=1.1.0
-    # extra: dmypy
-    - psutil >=4.0
 
-test:
-  imports:
-    - mypy
-    - mypy.plugins
-    - mypy.server
-    - mypy.dmypy
-  commands:
-    - mypy --help
-    - stubgen --help
-    - dmypy --help
-    - stubtest --help
-    - dir %SP_DIR%\mypy\*.pyd /s /b  # [win]
-    - ls $SP_DIR/mypy/*.so  # [linux and not (aarch64 and py==310)]
+outputs:
+  - name: mypy
+    build:
+      script:
+        - export MYPY_USE_MYPYC=1  # [not win]
+        - export SETUPTOOLS_USE_DISTUTILS=stdlib  # [not win]
+        - set MYPY_USE_MYPYC=1     # [win]
+        - set SETUPTOOLS_USE_DISTUTILS=stdlib  # [win]
+        # mypy searches for <>-stubs in site.getsitepackages()
+        # which points to $SP_DIR, even in cross-python
+        - cp -rv $BUILD_PREFIX/lib/python$PY_VER/site-packages/*-stubs $SP_DIR  # [build_platform != target_platform]
+        - cp -rv $BUILD_PREFIX/lib/python$PY_VER/site-packages/tomli* $SP_DIR  # [build_platform != target_platform]
+        - {{ PYTHON }} -m pip install . --no-deps -vv
+      entry_points:
+        - mypy = mypy.__main__:console_entry
+        - stubgen = mypy.stubgen:main
+        - dmypy = mypy.dmypy.client:console_entry
+        - stubtest = mypy.stubtest:main
+        # CLI provided here, but doesn't depend on a compiler
+        - mypyc = mypyc.__main__:main
+
+    requirements:
+      build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+        - {{ compiler('c') }}
+        # mypy's build imports this module, which has a C extension,
+        # so it has to be native to the build platform
+        - typed-ast >=1.4.0,<2                   # [build_platform != target_platform and py<38]
+        - tomli >=1.1.0
+      host:
+        - python
+        - pip
+        - mypy_extensions >=0.4.3
+        - typed-ast >=1.4.0,<2  # [py<38]
+        - typing_extensions >=3.10
+        - types-setuptools
+        - types-psutil
+        - types-typed-ast >=1.5.8,<1.6.0
+        - tomli >=1.1.0
+      run:
+        - python
+        - mypy_extensions >=0.4.3
+        - typed-ast >=1.4.0,<2  # [py<38]
+        - typing_extensions >=3.10
+        - tomli >=1.1.0
+        # extra: dmypy
+        - psutil >=4.0
+
+    test:
+      requires:
+        - pip
+      imports:
+        - mypy
+        - mypy.dmypy
+        - mypy.plugins
+        - mypy.server
+        - mypyc
+      commands:
+        - pip check
+        - dmypy --help
+        - mypy --help
+        - mypyc --help
+        - stubgen --help
+        - stubtest --help
+        - dir %SP_DIR%\mypy\*.pyd /s /b  # [win]
+        - ls $SP_DIR/mypy/*.so  # [linux and not (aarch64 and py==310)]
+
+  - name: mypyc
+    requirements:
+      host:
+        - python
+      run:
+        - {{ pin_subpackage("mypy", exact=True) }}
+        - c-compiler
```

Comment:
Think we want this

```suggestion
        - {{ compiler("c") }}
```

Reply:
ah, had been working off https://github.com/conda-forge/mypy-feedstock/issues/83#issuecomment-1308711145. will happily defer to whatever's best, especially if we find a case where it's busted... have only been able to get test reports locally on linux, despite cajoling folk with windows/apple boxen...

Reply:
related: https://github.com/conda-forge/staged-recipes/pull/21144

Reply:
Submitted PR ( https://github.com/conda-forge/mypy-feedstock/pull/87 )